scripts/lib/CIME/case/case.py

```diff
@@ -530,6 +533,71 @@ def _find_primary_component(self):
         return primary_component
 
 
+    __mod_match_re__ = re.compile(r"([^%]*[^0-9%]+)")
+    def _valid_compset(self, files, compset_name):
+        "Fill in missing stub models, return full compset name"
+        drv_config_file = files.get_value("CONFIG_CPL_FILE")
+        drv_comp = Component(drv_config_file, "CPL")
+        comp_classes = drv_comp.get_valid_model_components()
+        model_hash = {} # Hash model name to component class index
+        model_set = [None]*len(comp_classes)
+        # First, create hash of model names
+        # A note about indexing. Relevant component classes start at 1
+        # because we ignore CPL for finding model components.
+        # Model components would normally start at zero but since we are
+        # dealing with a compset, 0 is reserved for the time field
+        for comp_ind in xrange(1, len(comp_classes)):
+            comp = comp_classes[comp_ind]
+            # Find list of models for component class
+            # List can be in different locations, check CONFIG_XXX_FILE first
+            node_name = 'CONFIG_{}_FILE'.format(comp)
+            models = files.get_components(node_name)
+            if (models is None) or (None in models):
+                # Backup, check COMP_ROOT_DIR_XXX
+                node_name = 'COMP_ROOT_DIR_' + comp
+                models = files.get_components(node_name)
```

Comment:
Backup, look where CESM lists supported components.

Reply:
Would it help to have a bit of error checking here - e.g., abort if models is still None (and/or maybe if (None in models)??? I don't totally understand what is an error condition and what's okay here)